### ISSUE TITLE
Web siders are sorted by length, but not by name.

### ISSUE THREAD

**Issue by Baltazar500** [7sDream/torrent_parser#7]
Hi.

Web siders are sorted by length, but not by name, which makes the list of files turn into a dump of strings. Transmission-show sorts files by name, but its output is very hard to parse

Example

>Through.The.Wormhole.1х06.avi
>Through.The.Wormhole.1х08.avi
>Through.The.Wormhole.1х04.avi
>Through.The.Wormhole.1х01.avi
>Through.The.Wormhole.1х05.avi
>Through.The.Wormhole.1х02.avi
>Through.The.Wormhole.1х03.avi
>Through.The.Wormhole.1х07.avi

In transmission-show or torrent_parser + sort utility

>Through.The.Wormhole.1х01.avi
>Through.The.Wormhole.1х02.avi
>Through.The.Wormhole.1х03.avi
>Through.The.Wormhole.1х04.avi
>Through.The.Wormhole.1х05.avi
>Through.The.Wormhole.1х06.avi
>Through.The.Wormhole.1х07.avi
>Through.The.Wormhole.1х08.avi

**Comment by 7sDream** [comment]
I don't quite understand what you mean and the problem you faced, but it seems something about fields' ordering in decoded dict? 

I will try give you some solutions about the problem I guess, but if I guess wrong, fell free to provide more info and describe your question.

---

Are you using `pytp` CLI or use `torrent_parser` as a Python module?

If you're using `pytp`, it will by default use `OrderedDict`, and keeps fields' ordering. If you meet this problem when using `pytp`, please provide the torrent file you use and I will look into it.

If you're writing code using `torrent_parser` module, in fact [`parse_torrent_file`](https://github.com/7sDream/torrent_parser/blob/2a4eecb83081154e4a0341633ab05f65a7aa7575/torrent_parser.py#L566) function, [`TorrentFileParser`](https://github.com/7sDream/torrent_parser/blob/2a4eecb83081154e4a0341633ab05f65a7aa7575/torrent_parser.py#L144) class and [`BDecoder`](https://github.com/7sDream/torrent_parser/blob/2a4eecb83081154e4a0341633ab05f65a7aa7575/torrent_parser.py#L488) class both has a `use_ordered_dict` argument, set it to `True` will make them use `OrderedDict` too.

**Comment by Baltazar500** [comment]
>Are you using pytp CLI or use torrent_parser as a python module?

I am using pytp under bash shell.

>cat ./WormHole.torrent|pytp -a|jq -r '(.httpseeds[] + .info.name + "/" + .info.files[].path[])'
>pytp -a ./WormHole.torrent|jq -r '(.httpseeds[] + .info.name + "/" + .info.files[].path[])'

>If you're using pytp, it will by default use OrderedDict, and keeps fields' ordering. If you meet this problem when using pytp, please provide the torrent file you use and I will look into it.

https://www.upload.ee/files/11499058/Wormhole.torrent.html

>If you're writing code use torrent_parser module, in fact parse_torrent_file function, TorrentFileParser class and BDecoder class both has a use_ordered_dict argument, set it to True will make them use OrderedDict too.

I do not know how to create scripts in python :(

**Comment by 7sDream** [comment]
It's OK~

I'll check the torrent file on weekends.

**Comment by 7sDream** [comment]
![image](https://user-images.githubusercontent.com/7822577/79631389-b2e4d580-818b-11ea-8158-0622d2be2d06.png)

Yes, I can reproduce your result.

And it seems because of the torrent creator do not sort file by it's path, so, as a simple parser, pytp can only choose to keep the origin item order in a list……

So I'm sorry to say I can't do anything to solve this. But you can continue use `sort` util tool if you only want to get file names.

**Comment by Baltazar500** [comment]
>So I'm sorry to say I can't do anything to solve this. But you can continue use sort util tool if you only want to get file names.

ОК.
